src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3838,17 +3838,32 @@ private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter)
                 type = null;
                 name = this.EatToken(SyntaxKind.ArgListKeyword);
             }
-
+            SyntaxToken exclamation = null;
+            SyntaxToken equals = null;
+            if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
+            {
+                exclamation = this.EatToken(SyntaxKind.ExclamationToken);
+            }
+            else if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken)
```

Comment:
>else if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken) [](start = 12, length = 69)

From an offline discussion we've decided to treat `!=` as an error but interpreted semantically as a null-checking token followed by an optional value.

The resulting ParameterSyntax instance would have a missing ExclamationToken, and an EqualsValueClause where the `!=` token is used as the EqualsToken. The error would be attached to the `!=` (that is, to the EqualsValueClause.Equals token).

The reason for treating `!=` as an error rather than splitting the token into two is to avoid complexity with incremental parsing.

Reply:
@cston That works for me.  Note: i'm fairly certain there wouldn't be an incremental parsing issue here since one of htese tokens would hav ea diagnostic on it (which prevents incremental).  But i have no issue just with keeping things simple and conservative to make life easier :)